rust/azure_iot_operations_mqtt/src/session/managed_client.rs

```diff
@@ -43,35 +51,76 @@ impl SessionManagedClient {
         SessionPubReceiver { pub_rx }
     }
 
+    /// Creates a new [`SessionPubReceiver`] that will receive all incoming publishes that are NOT
+    /// sent to any filtered receivers.
+    ///
+    /// If you want to receive ALL publishes, use a filtered receiver with a wildcard topic (#).
+    ///
+    /// # Panics
+    /// Panics if internal state is invalid (this should not be possible).
+    #[must_use]
     pub fn create_unfiltered_pub_receiver(&self) -> SessionPubReceiver {
         let pub_rx = self.dispatcher.lock().unwrap().create_unfiltered_receiver();
         SessionPubReceiver { pub_rx }
     }
 
+    /// Issue an MQTT `PUBLISH` at Quality of Service 0 ("at most once" delivery).
+    ///
+    /// If connection is unavailable, `PUBLISH` will be queued and delivered when connection is
+    /// re-established. Blocks if at capacity for queueing.
+    ///
+    /// Returns a token that can be awaited to indicate the completion of the `PUBLISH` operation
+    /// (i.e. when the `PUBLISH` has been sent to the server).
+    ///
+    /// # Errors
+    /// Returns a [`DetachedError`] if the `PUBLISH` could not be issued due to being detached from
+    /// the Session
     pub async fn publish_qos0(
         &self,
         topic: TopicName,
         retain: bool,
         payload: impl Into<Bytes> + Send,
         properties: PublishProperties,
-    ) -> Result<PublishQoS0CompletionToken, ClientError> {
+    ) -> Result<PublishQoS0CompletionToken, DetachedError> {
         self.client
             .publish_qos0(topic, payload.into(), retain, properties)
             .await
     }
 
+    /// Issue an MQTT `PUBLISH` at Quality of Service 1 ("at least once" delivery).
+    ///
+    /// If connection is unavailable, `PUBLISH` will be queued and delivered when connection is
+    /// re-established. Blocks if at capacity for queueing.
+    ///
+    /// Returns a token that can be awaited to indicate the completion of the `PUBLISH` operation
```

Comment:
doesn't the first await correlate to the publish being sent and the second one to the puback being received? Not sure if there's anything around that that should be clarified in this comment so that folks make sure to .await twice if they want full confirmation